pkg/csi/cinder/nodeserver.go

```diff
@@ -540,6 +541,17 @@ func (ns *nodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	if _, err := r.Resize(devicePath, volumePath); err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not resize volume %q:  %v", volumeID, err)
 	}
+
+	newFsSize, err := filesystem.GetFilesystemSize(volumePath)
```

Comment:
You may want to use `r.NeedResize()` instead:

https://github.com/kubernetes/mount-utils/blob/master/resizefs_linux.go#L105

mount-utils is added in this PR: https://github.com/kubernetes/cloud-provider-openstack/pull/1440, but it's interesting it doesn't exist in this file 🤔

It can also be used for https://github.com/kubernetes/cloud-provider-openstack/issues/1539 as well :)

Reply:
I thought I've sent it as a comment, not a review! :D

Reply:
Aye, but the problem is. Due to the bug that I linked, the block device won't be resized as well. I think I'll add a check for the block device size; let's see how it works out. And if I can persuade maintainers to merge this.